hal/api/toolchain.h

```diff
@@ -225,6 +225,35 @@
 #endif
 #endif
 
+/* Define '__RAMFUNC' as an attribute to mark a function as residing in RAM.
+ * Use of __RAMFUNC puts a function in the .data section--i.e. the
+ * initialized data section. This will be copied into RAM automatically by the
+ * startup sequence. */
+#ifndef __RAMFUNC
+#if defined(__GNUC__) || defined(__clang__) // GCC and llvm/clang
+#define __RAMFUNC __attribute__ ((section (".data#"), noinline)) /* The '#' following ".data" needs a bit of
```

Comment:
Actually, a completely separate name is the right thing to do for ARMCC too. You were previously worried about the issue of actually marking RAM sections executable. You do want some special naming to allow grouping together into an executable region if necessary.

Lumping this stuff in ".data.*" is just asking for it to end up in a non-executable region.